File: Order.cs
/*
*
* The whole info about deserialising json has been taken from here
* https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-how-to?pivots=dotnet-6-0
*
*/
class Order
{
    public int order_id;
    public int table_id;
    public int waiter_id;
    public List<int> items;
    public int priority;
    public int max_wait;
    public int pickup_time;
    public int cooking_time;
    public List<Dictionary<String, int>>? cooking_details;
    public Order()
    {
        items = new List<int>();
    }
    public Order(Order a)
    {
        order_id = a.order_id;
        table_id = a.table_id;
        waiter_id = a.waiter_id;
        memberw


    }
}

File: order.go
package main

import (
	"log"
	"math"
)

type Order struct {
	OrderId        int              `json:"order_id,omitempty"`
	TableId        int              `json:"table_id,omitempty"`
	WaiterId       int              `json:"waiter_id,omitempty"`
	Items          []int            `json:"items,omitempty"`
	Priority       int              `json:"priority,omitempty"`
	MaxWait        int              `json:"max_wait,omitempty"`
	PickUpTime     int64            `json:"pick_up_time,omitempty"`
	CookingTime    int              `json:"cooking_time,omitempty"`
	CookingDetails []map[string]int `json:"cooking_details,omitempty"`
}

func NewOrder(orderId, tableId, waiterId int, items []int, priority, maxWait int, pickUpTime int64) *Order {
	return &Order{
		OrderId:    orderId,
		TableId:    tableId,
		WaiterId:   waiterId,
		Items:      items,
		Priority:   priority,
		MaxWait:    maxWait,
		PickUpTime: pickUpTime}
}

func (o *Order) getMaxPrepTime() {
	maxPrepTime := 0
	for _, item := range o.Items {
		if CurrentMenu[item].PreparationTime > maxPrepTime {
			maxPrepTime = CurrentMenu[item].PreparationTime
		}
	}
	o.MaxWait = int(math.Ceil(1.3 * float64(maxPrepTime)))
	log.Printf("Max prep time :%v\n", maxPrepTime)
}
